data/mods/apply_mods.py

```diff
@@ -64,9 +64,40 @@ def apply_gen_6_mods():
 def apply_gen_7_mods():
     apply_move_mods(7)
     apply_pokedex_mods(7)
+    
+PRE_PHYSICAL_SPECIAL_SPLIT_CATEGORY_LOOKUP = {
+    "normal": constants.PHYSICAL,
+    "fighting": constants.PHYSICAL,
+    "flying": constants.PHYSICAL,
+    "poison": constants.PHYSICAL,
+    "ground": constants.PHYSICAL,
+    "rock": constants.PHYSICAL,
+    "bug": constants.PHYSICAL,
+    "ghost": constants.PHYSICAL,
+    "steel": constants.PHYSICAL,
+    "fire": constants.SPECIAL,
+    "water": constants.SPECIAL,
+    "grass": constants.SPECIAL,
+    "electric": constants.SPECIAL,
+    "psychic": constants.SPECIAL,
+    "ice": constants.SPECIAL,
+    "dragon": constants.SPECIAL,
+    "dark": constants.SPECIAL,
+}
```

Comment:
This lookup is missing `fairy`. The type didn't exist in gen3, but below the code is:

```python3
PRE_PHYSICAL_SPECIAL_SPLIT_CATEGORY_LOOKUP[move_data[constants.TYPE]]
```

This will surely `KeyError` for fairy moves.

Reply:
Alternatively, you can change the code below to be:

```python3
try:
  move_data[constants.CATEGORY] = PRE_PHYSICAL_SPECIAL_SPLIT_CATEGORY_LOOKUP[move_data[constants.TYPE]]
except KeyError:
  pass
```